File: pset6/sentimental-readability/readability.py
# TODO
import re


def main():

    try:
        text = input('Text: ')
        cli = cl_index(text)
    except Exception as e:
        print(e)

    if cli > 16:
        print('Grade 16+')
    elif cli < 1:
        print('Before Grade 1')
    else:
        print('Grade', cli)

    exit()


def cl_index(text):

    if not len(text.split()):
        exit(1)

    letters = len(re.findall('\w', text))   # Letter counter
    words = len(text.split())       # Word counter
    sentences = len(re.findall('\.|\!|\?', text))   # Count sentences by counting the number of ". ! ?"

    l = float(float(letters / words) * 100)     # Average letters per 100 words
    s = float(float(sentences / words) * 100)   # Average sentences per 100 words
    cli = 0.0588 * l - 0.296 * s - 15.8     # Coleman-Liau index formula

    return round(cli)   # Return rounded index to the nearest integer


if __name__ == '__main__':
    main()

File: pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


int count_letters(string text);
int count_words(string text);
int count_phrases(string text);
int main(void)
{
    string text1 = get_string("Text: ");
    float index = 0;
    int letters = count_letters(text1);
    int words = count_words(text1);
    int phrases = count_phrases(text1);
    float l = (float) letters / (float) words * 100.0;
    float s = (float) phrases / (float) words * 100.0;
    index = 0.0588 * l - 0.296 * s - 15.8;  //Coleman-Liau index
    int grade = roundf(index);              //round index to an int

    if (index < 1)                          //if grade is below 1
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)                    //if grade is above 16
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);        //every other grade
    }
}

int count_letters(string text)              //count number of letters
{
    int letters = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (isalpha(text[i]))               //counter for every alphabetic char
        {
            letters++;
        }
    }
    return letters;
}

int count_words(string text)                //count number of words, using spaces for reference
{
    int words = 1;                           //int words is set to one, to count the first word
    for (int i = 0, len = strlen(text); i < len ; i++)  //be fore the first space
    {
        if (isspace(text[i]))
        {
            words++;
        }
    }
    if (words == 1)     //int words is set to 1, to prevent counting extra word if only one word is typed
    {
        words--;
    }
//    printf("%i \n", words);
    return words;
}

int count_phrases(string text)              //count sentences by counting . ! ?
{
    int phrases = 0;
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if (text[i] == '.')
        {
            phrases++;
        }
        if (text[i] == '!')
        {
            phrases++;
        }
        if (text[i] == '?')
        {
            phrases++;
        }
    }
//    printf("%i \n", phrases);
    return phrases;
}